### PR TITLE
Add multiple flags to reduce description string duplication

### PR DESCRIPTION
**Pull request by haveric**
#### Summary

Infrastructure "Add multiple flags to reduce description string duplication"

#### Purpose of change

- Reduce duplication of strings that make sense within the flags section. Related to #55821
- Fixes irradiated celery and irradiated cucumber stats (weren't copying from originals)
- Minor grammar fixes in docs.

#### Describe the solution

Added seven new flags and updated items to use them instead of duplicating text across items.

#### Describe alternatives you've considered

- Parse descriptions by sentence for the translator.
- Allow descriptions to be an array of sentences which each get translated separately.
- Create a new type of flag that appends to the description.

#### Testing

Loaded the game with no errors. Looked through some of the items in debug mode.

#### Additional context

image
image
image
image
image

### CODE REVIEW

**Review by LyleSY — APPROVED** [review]

**Comment by NetSysFire** [comment]
Looks fine so far, but your last two screenshots showing the mutagenic primers have "you need a syringe" twice in their attributes.

**Comment by haveric** [comment]
> mutagenic primers have "you need a syringe" twice in their attributes.

Interesting. It must be pulling the activate component already. I'll take a look to see if I can just remove that entirely then.

**Comment by haveric** [comment]
@NetSysFire I've simply removed my NEED_SYRINGE. There's code applying the other message based on the use_action having a syringe, so there's no need to duplicate it.

File: data/json/items/ammo/20x66mm.json [hunk 1]
```diff
@@ -33,50 +33,50 @@
...
     "name": { "str_sp": "20x66mm explosive" },
-    "description": "20x66mm caseless shotgun rounds, explosive projectile type.  Proprietary ammunition for Rivtech shotguns.  Being caseless rounds, these cannot be disassembled or reloaded.",
+    "description": "20x66mm caseless shotgun rounds, explosive projectile type.  Proprietary ammunition for Rivtech shotguns.",
     "price": 10000,
     "price_postapoc": 8000,
     "range": 20,
     "damage": { "damage_type": "bullet", "amount": 16, "armor_penetration": 4 },
     "dispersion": 120,
-    "effects": [ "COOKOFF", "EXPLOSIVE_20x66", "NEVER_MISFIRES" ]
+    "extend": { "effects": [ "EXPLOSIVE_20x66" ], "flags": [ "CASELESS_ROUNDS" ] }
```
Comment (Faalagorn) [inline data/json/items/ammo/20x66mm.json:72]: I just realized it; was removal of `COOKOFF` and `NEVER_MISFIERES` flag intentional @haveric?

Comment (haveric) [reply data/json/items/ammo/20x66mm.json:72]: It's extending the abstract which includes those: https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/items/ammo/20x66mm.json#L15

**Pull request merged**